### PR TITLE
Docs: Added snes9x-nwa as recommended emulator to the setup guides for SNES games

### PR DESCRIPTION
**Pull request by CubeSoldier**
## What is this fixing or adding?
This adds a new variant of snes9x to the list of emulators detailed in the setup guides for SNES games. The main advantage of this version compared to snes9x-rr is that the NWA protocol this uses doesnt require any usage of Lua scripting and is fully automatic on each subsequent boot of the emulator.  In addition its based on Snes9x 1.61 instead of 1.60.

So overall this should reduce tech issues related to Lua Scripting and other Lua Bridge DLL Shenanigans. If approved and tested further this could completely replace snes9x-rr as a recommended emulator completely.

However this Emulator should only be used with SNI versions 0.0.90 or later as I found during my testing that SNI might crash or lock up on older versions while using this emulator. So if youre going to merge this only merge it when also bundling the latest version of SNI.

## How was this tested?
Tested it myself during multiple Archipelago Multiworld Seeds of SMZ3, Super Metroid and ALttP while using SNI Version 0.0.90

### CODE REVIEW

**Review by black-sliver — APPROVED** [review]
We bundle the latest SNI when creating the installer, so that would happen at the same time this doc does up.

The advantages of snes9x-nwa is that it could get mac and linux builds down the road, while this seems unrealistic for -rr and -multitroid, and it avoids typical user error. The only downside I'm aware of is that it is potentially not thread-safe, so there is a chance problems creep up with certain games that rely on emulator behaviour and I'll leave it up to @Berserker66 to hit the merge button.

**Comment by Berserker66** [comment]
"The only downside I'm aware of is that it is potentially not thread-safe, so there is a chance problems creep up with certain games that rely on emulator behaviour"
Great.
I keep hearing bad things about nwa, it seems the only upside is ease of use and then it just goes downhill from there.

**Comment by Berserker66** [comment]
Should then probably ping every affected worlds maintainers to verify it does actually work?

**Comment by black-sliver** [comment]
sorry for the late reply. the kind of thread-unsafely should be comparable to what we have with fxpak. the reason i brought it up is that bizhawk and RA behave differently and i am not sure how well-tested some of the games are on fxpak

**Comment by Berserker66** [comment]
Yeah, I might be the wrong person for that then. LttP works fine on fxpak, as long as one uses SNI.

**Comment by Berserker66** [comment]
image
So, that sounds really bad.

**Comment by black-sliver** [comment]
The rewind issue was solved. The EmuNWA protocol seems to find acceptance over in other discords, because it's easier to use. Should maybe still hold off a bit to slowly test and roll out the updated Snes9X, but it looks promising.

More testing with other games (like DKC3, SMW, ...) would be appreciated since I really don't know how well they work with either FxPak or EmuNWA (which behave similar).

**Comment by CubeSoldier** [comment]
Hey I had someone test this for me with SMW as part as a multiworld and it worked completely fine. No issues whatsoever during two full playthroughs. So this is ALTTP, SM, SMZ3 and SMW tested fine so I'll assume it should work fine as well with the remaining SNES games.

**Comment by ThePhar** [comment]
Not sure why we haven't pinged other relevant world owners yet. Adding these to their docs would open them up to support questions about those emulators so I want their opinions before making a general approval on this change.

@PoryGone for DKC3 and SMW
@lordlou for SM and SMZ3

Would also like your approval @Berserker66 if it's fine for ALTTP's docs.

**Comment by lordlou** [comment]
I've never tried nwa neither but I'm not against adding it if it works well. I would be interrested to know if EmoTracker works with it and SMZ3 though.

**Comment by black-sliver** [comment]
> I've never tried nwa neither but I'm not against adding it if it works well. I would be interrested to know if EmoTracker works with it and SMZ3 though.

SNI simply provides it as an alternative back-end, so if a tracker supports the usb2snes protocol, it should just work.

**Comment by CubeSoldier** [comment]
> I've never tried nwa neither but I'm not against adding it if it works well. I would be interrested to know if EmoTracker works with it and SMZ3 though.

It definitely works with Emotracker just using the USB2Snes protocol. I've used the Emotracker SMZ3 Pack for my AP Test Runs.

> I've never touched nwa, and cannot vouch for it even functioning for my games (DKC3 and SMW), so I would prefer that it not be listed for those.

I and a friend of mine successfully ran multiple SMW seeds using it with no issues whatsoever.

**Comment by CubeSoldier** [comment]
> > A friend of mine and I successfully ran multiple SMW seeds using it with no issues whatsoever.
> 
> This information does not change my opinion. Please remove those two games from this PR.

Done. This only leaves Secret of Evermore as the only untested game.

**Comment by black-sliver** [comment]
I've been using emunwa interface with evermore in my bsnes-plus fork. While not having tested the snes9x implementation, it should work.

File: worlds/alttp/docs/multiworld_en.md
Comment (alwaysintreble) [inline worlds/alttp/docs/multiworld_en.md:9]: slight grammar suggestion. I use bsnes-plus for lttp, so also added that here if you'd like it. installation process to get it running is even less than snes9x-nwa. the networking is always enabled so it just works as is, if you want to add a blurb for it below.
```suggestion
- Hardware or software capable of loading and playing SNES ROM files, including:
    - An emulator capable of connecting to SNI
      ([snes9x rr](https://github.com/gocha/snes9x-rr/releases),
       [BizHawk](https://tasvideos.org/BizHawk), or
      ([snes9x-nwa](https://github.com/Skarsnik/snes9x-emunwa/releases), [snes9x rr](https://github.com/gocha/snes9x-rr/releases),
       [BSNES-plus](https://github.com/black-sliver/bsnes-plus),
       [BizHawk](http://tasvideos.org/BizHawk.html), or
       [RetroArch](https://retroarch.com?page=platforms) 1.10.1 or newer)
    - An SD2SNES, [FXPak Pro](https://krikzz.com/store/home/54-fxpak-pro.html), or other compatible hardware. **note: 
modded SNES minis are currently not supported by SNI. Some users have claimed success with QUsb2Snes for this system,
but it is not supported.**
```

**Comment by black-sliver** [comment]
@CubeSoldier can you react to treble's suggestions?

--
Small update on snex9x-nwa: I only heard of a singular crash in snes9x-nwa and no tracking issues, so it seems to be rather stable.

**Comment by CubeSoldier** [comment]
Sorry I was kinda busy over the holidays. Ive been personally using Snes9x-nwa for all my multiworld runs now and it seems stable. I have no qualms adding BSNES plus as well as that fork is developed by the ALTTPO team anyways.

File: worlds/alttp/docs/multiworld_en.md
Comment (black-sliver) [inline worlds/alttp/docs/multiworld_en.md:23]: hm, that section got duplicated. I think the suggestion was not meant to be applied as-is. :/
Once that's fixed I'm gonna merge it and we can revert if people find bugs.

Comment (CubeSoldier) [reply worlds/alttp/docs/multiworld_en.md]: Done. My apologies for missing this. I also removed the duplicate mentions of snes9x-rr and added the missing dash for it.

**Comment by black-sliver** [comment]
@alwaysintreble can you double-check if this retains your suggestion now?

**Comment by alwaysintreble** [comment]
I was hoping the suggestion there would be applied to the other docs too, and it also needs a blurb about how to connect bsnes like how all the other emulators do. while it does work out of the box, the fact that the blurb isn't there will just leave players confused.

**Review by alwaysintreble — APPROVED** [review]

**Review by black-sliver — APPROVED** [review]
Thanks!

Haven't really gone over the final version, but all files' section now look the same and I believe treble did.

**Pull request merged**
